xarray/core/groupby.py

```diff
@@ -71,9 +71,27 @@ def check_reduce_dims(reduce_dims, dimensions):
             raise ValueError(
                 f"cannot reduce over dimensions {reduce_dims!r}. expected either '...' "
                 f"to reduce over all dimensions or one or more of {dimensions!r}."
+                f" Try passing .groupby(..., squeeze=False)"
             )
 
 
+def _maybe_squeeze_indices(
+    indices, squeeze: bool | None, grouper: ResolvedGrouper, warn: bool
+):
+    if squeeze in [None, True] and grouper.can_squeeze:
+        if isinstance(indices, slice):
+            if indices.stop - indices.start == 1:
```

Comment:
Is that save?
What about None or negative Start-Stop Values?

Reply:
Both are not possible today:
` group_indices: T_GroupIndices = [slice(i, i + 1) for i in range(size)]` at Line 455

and at Line 553:
```
group_indices: T_GroupIndices = [
            slice(i, j) for i, j in zip(sbins[:-1], sbins[1:])
]
```

Reply:
Ah no you're right, there's an edge case in TimeResamplerGrouper where `stop` is None, and we're resampling to the same frequency as the data so `grouper.can_squeeze` is True.

Reply:
Ah no it doesn't run, but i have no explicitly skipped squeezing for resampling. This is the current behaviour on `main`